reflect func conv error

```
package main

import (
	"reflect"
	"strconv"
)

type itoaFunc func(i int) string

func (f itoaFunc) Itoa(i int) string { return f(i) }

func main() {
	typ := reflect.TypeOf((*itoaFunc)(nil)).Elem()
	tyString := reflect.TypeOf("")
	v := reflect.MakeFunc(typ, func(args []reflect.Value) []reflect.Value {
		r := strconv.Itoa(int(args[0].Int()))
		return []reflect.Value{reflect.ValueOf(r)}
	})
	ftyp := reflect.FuncOf([]reflect.Type{v.Type()}, []reflect.Type{tyString}, false)
	fn := reflect.MakeFunc(ftyp, func(args []reflect.Value) []reflect.Value {
		r := args[0].Call([]reflect.Value{reflect.ValueOf(100)})
		return r
	})
	r := fn.Call([]reflect.Value{v})
	if r[0].String() != "100" {
		panic("func conv error: " + r[0].String())
	}
}
```

```
panic: func conv error: 0
```

--- a/.github/workflows/test_demo.sh
+++ b/.github/workflows/test_demo.sh
@@ -100,6 +100,7 @@ ignore_esp32=(
   "./_demo/go/readdir" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectcopy" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectfunc" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectfnconv" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectindirect" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectmake" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectmethod" # panic: internal/bytealg selected .s files require plan9asm translation
@@ -173,6 +174,7 @@ ignore_esp32c3_basic=(
   "./_demo/go/randdemo" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/readdir" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectfunc" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectfnconv" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectindirect" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectcopy" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectmethod" # panic: internal/bytealg selected .s files require plan9asm translation

--- a/_demo/go/reflectfnconv/main.go
+++ b/_demo/go/reflectfnconv/main.go
@@ -1,0 +1,28 @@
+package main
+
+import (
+	"reflect"
+	"strconv"
+)
+
+type itoaFunc func(i int) string
+
+func (f itoaFunc) Itoa(i int) string { return f(i) }
+
+func main() {
+	typ := reflect.TypeOf((*itoaFunc)(nil)).Elem()
+	tyString := reflect.TypeOf("")
+	v := reflect.MakeFunc(typ, func(args []reflect.Value) []reflect.Value {
+		r := strconv.Itoa(int(args[0].Int()))
+		return []reflect.Value{reflect.ValueOf(r)}
+	})
+	ftyp := reflect.FuncOf([]reflect.Type{v.Type()}, []reflect.Type{tyString}, false)
+	fn := reflect.MakeFunc(ftyp, func(args []reflect.Value) []reflect.Value {
+		r := args[0].Call([]reflect.Value{reflect.ValueOf(100)})
+		return r
+	})
+	r := fn.Call([]reflect.Value{v})
+	if r[0].String() != "100" {
+		panic("func conv error: " + r[0].String())
+	}
+}

--- a/runtime/internal/lib/reflect/makefunc.go
+++ b/runtime/internal/lib/reflect/makefunc.go
@@ -136,6 +136,9 @@ func storeMakeFuncResult(ret unsafe.Pointer, v Value, typ *abi.Type) {
 
 func ffiToValue(ptr unsafe.Pointer, typ *abi.Type) (v Value) {
 	kind := typ.Kind()
+	if typ.Kind() == abi.Func {
+		typ = closureOf(typ.FuncType())
+	}
 	v.typ_ = typ
 	v.flag = flag(kind)
 	if typ.IfaceIndir() {

--- a/runtime/internal/lib/reflect/value.go
+++ b/runtime/internal/lib/reflect/value.go
@@ -589,6 +589,8 @@ func (v Value) Elem() Value {
 		kind := typ.Kind()
 		if typ.IsClosure() {
 			kind = abi.Func
+		} else if kind == abi.Func {
+			typ = closureOf(typ.FuncType())
 		}
 		fl |= flag(kind)
 		return Value{typ, ptr, fl}

